api/src/features/component/component_feature.py

```diff
@@ -5,11 +5,12 @@
 
 from common.components import COMPONENT_IDS
 from common.responses import create_response
+from features.component.component_use_case import component_use_case
 
 router = APIRouter(tags=["component"])
 
 
 @create_response(JSONResponse)
-@router.get("/components", operation_id="get_components", response_model=Dict[int, Dict[str, str]])
-async def components() -> Dict[int, Dict[str, str]]:
-    return COMPONENT_IDS
+@router.get("/components", operation_id="get_components", response_model=Dict[str, Dict[str, str]])
+async def components() -> Dict[str, Dict[str, str]]:
+    return component_use_case(COMPONENT_IDS)
```

Comment:
For the TS generated types, a BaseModel class for the response is needed.
`components() -> ComponentIds:
    return ComponentsIds(**COMPONENT_IDS)

Would also make this use_case unnecessary

Reply:
fixed